apis/nucleus/src/components/listbox/ListBox.jsx

```diff
@@ -151,7 +151,10 @@ export default function ListBox({
                 '/qListObjectDef',
                 sorted.map((s) => ({
                   qTop: s.start,
-                  qHeight: s.stop - s.start + 1,
+                  // we always need to ask for MINUM_BATCH_SIZE
+                  // if backend sens anything less than that
+                  // then we know that is last page for us
+                  qHeight: MINIMUM_BATCH_SIZE,
```

Comment:
Does that also work in analytics mode?

Reply:
will check it with analytics team asap

Reply:
I tested it, seems working. A little bit weird when for example searching for a specific value, `s.stop - s.start + 1` is 1, but now we are asking for 100 rows in a page.

Reply:
do we get the correct result from api as long as we have less items but still asking for 100 items? 
in my testings Yes it does because my thinking is that regardless of asking for 100: 
  1. backend would send up what ever it has (it could be less than 100)
  2. based on that amount of items, we recognise that if we are in last page or not
  3. then we try to calculate height for last page and

 all of this chain works together 👌! by causing **no** issues on both sides!

Reply:
When I debug it, it looks like even set` qHeight: MINIMUM_BATCH_SIZE,` the next chain `p` get exact number of values should return. I am not sure if necessary to change to ` qHeight: MINIMUM_BATCH_SIZE,`.  
![Screenshot 2022-07-22 at 13 31 09](https://user-images.githubusercontent.com/25456307/180430678-a880c45d-c494-475c-beaa-a2394317040c.png)

Reply:
that is because either backend has sent you the last page of results or you only have that 2 items in your resulted items, I assume you've done some filtering/searching through the list and regardless of asking for 100 items, engine sends you the all-existing/last 2 items that it has for that searching result

👆 i think this is another sign that asking for 100 items will not cause any issues in analytics mode!